Adds the players "high" job choice next to their name in the lobby

Nero-07 opened the pull request:
I tried for like 5 hours to get it to automatically pull the job the player selected "high" for, but through an unfortunate combination of my own inexperience and confusing code, this is the best i could come up with.

https://baystation12.net/forums/threads/suggestion-roundstart-jobs-assignment.3013/

Nero-07 committed:
Adds a new prompt when players click ready in lobby

So they can specify what Job they are going to play

https://baystation12.net/forums/threads/suggestion-roundstart-jobs-assignment.3013/

Nero-07 committed:
Update new_player.dm

Nero-07 committed:
Update new_player.dm

Nero-07 commented:
Updated to ask "what job(s) do you have selected?" instead of "What Job(s) will you play"

DeeUnderscore commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -87,7 +90,7 @@
 			totalPlayersReady = 0
 			for(var/mob/new_player/player in player_list)
-				stat("[player.key]", (player.ready)?("(Playing)"):(null))
+				stat("[player.key]", (player.ready)?("(Playing) as [highjob]"):(null))
```
Why not use `player.client.prefs.job_high` instead?

You may have to add some extra checks like this:

``` dm
var/highjob
if(player.client && player.client.prefs.job_high)
    highjob = " as [player.client.prefs.job_high]"
stat("[player.key]", (player.ready)?("(Playing[highjob])"):(null))
```

I'm not sure if `client` will ever be `null` at this point, but I'm pretty sure the high job selection can be blank, so that check at least may be in order somewhere. Either way, a quick and dirty test of the above shows that it works.

Nero-07 replied:
god damnit. I tried every other possible permutation instead of player.client.prefs.job_high and it always refused to compile. I spent hours on this crap :( Guess I'll ask first next time

Techhead0 replied:
As an alternative to DeeUnderscore's suggestion, you could instead use `player.client.prefs.job_high` to find and cache `highjob` when they hit the ready button. (Down in `Topic()`)

Nero-07 committed:
Update new_player.dm

Nero-07 commented:
Updated once again to just pull the players choice of "high" jobslot next to their name and nothing if they don't have one.

Changed the title to reflect the new content of this request

Nero-07 committed:
Update new_player.dm

ZCaliber commented:
I love you.

atlantiscze commented:
I assume this shows only when the player is ready?

Furthermore, if you are up to expand it a bit, might want to add the name of character they are readying up as too. Just a random suggestion.

PsiOmegaDelta commented:
Just a minor note, but this does make some meta information more easily acquired. Someone readies up for a given role with a given character name and then no one with that role and/or name show up IG, strongly suggesting they've been recruited as an off-station antag.

Already sort of possible if you know which roles/characters someone tend to join as, but as stated this makes it easier for everyone.

Nero-07 commented:
Well, offstation antags don't really try to infiltrate the crew as their normal character anyway right?

Raiders, mercs, wizards etc alöl get a new identity anyway, so whats the harm in knowing the guy who was going to play captain is now an offstation antag? When you see the roundtype is ninja, you know theres going to be one anyways

PsiOmegaDelta commented:
What if the round type is Secret?

Nero-07 commented:
If you think it's a problem it could only be shown on non secret rounds?

Nero-07 commented:
Or we just leave it as is for the moment - without the character name. So if someone joins as a cargo technician and there are already 3 of those, maybe he was just bumped to another job.

PsiOmegaDelta commented:
Mostly just saying. Can be extended later.
